docs/source/guides/data.rst

```diff
@@ -393,6 +393,9 @@ option when using sequential data.
     to ``limit_{mode}_batches``, if it is set to 1.0 it will run for the whole dataset, otherwise it will throw an exception.
     Here ``mode`` can be train/val/test/predict.
 
+When iterable datasets are used, Lightning will pre-fetch 1 batch (in addition to the current batch) so it can detect
+when the training will stop and run validation if necessary.
```

Comment:
It can happen at any point in training:

https://github.com/PyTorchLightning/pytorch-lightning/blob/02ccd874b9eda885df7752710e605e259d579d9c/pytorch_lightning/loops/epoch/training_epoch_loop.py#L488-L489

Reply:
then maybe `when the validation loop needs to be run.`

Reply:
while the comment here is not wrong, the real reason we have the prefetching is actually to avoid starting a new epoch that doesn't have any batches left in the dataloader. Relying on a StopIteration check is not possible for this reason.